client/src/components/map/Map.tsx

```diff
@@ -1,106 +1,218 @@
-    const mapContainerRef = useRef<any>(null);
-    const mapRef = useRef<mapboxgl.Map | null>(null);
-
-    useEffect(() => {
-        if (mapRef.current) return;
-
-        mapRef.current = new mapboxgl.Map({
-            container: mapContainerRef.current,
-            //style: "mapbox://styles/mapbox/satellite-streets-v12",
-            style: "mapbox://styles/mapbox/light-v11",
-            center: [20.26, 67.845],
-            zoom: props.zoom || defaultZoom,
-            pitch: 40,
-            interactive: false
-        });
-    }, []);
-
-    useEffect(() => {
-        if (!mapRef.current) return;
-
-        const map = mapRef.current;
-
-        map.setZoom(props.zoom || defaultZoom);
-    }, [props.zoom]);
-
-    return (
-        <div className={props.className} ref={mapContainerRef} id="map" style={props.style}></div>
-    )
-}
+  const mapContainerRef = useRef<any>(null);
+  const mapRef = useRef<mapboxgl.Map | null>(null);
+  const markerRef = useRef<Marker | null>(null);
+
+  useEffect(() => {
+    if (mapRef.current) return;
+
+    mapRef.current = new mapboxgl.Map({
+      container: mapContainerRef.current,
+      //style: "mapbox://styles/mapbox/satellite-streets-v12",
+      style: "mapbox://styles/mapbox/light-v11",
+      center: [20.26, 67.845],
+      zoom: props.zoom || defaultZoom,
+      pitch: 40,
+      interactive: false,
+    });
+  }, []);
+
+  useEffect(() => {
+    if (props.coordinates && mapRef.current) {
+      // Remove existing marker if present
+      if (markerRef.current) {
+        markerRef.current.remove();
+      }
+
+      // Create and add a new marker at the passed coordinates
+      markerRef.current = new Marker()
+        .setLngLat([props.coordinates.lng, props.coordinates.lat])
+        .addTo(mapRef.current);
+
+      // Optionally, center the map on the new marker
+      mapRef.current.setCenter([props.coordinates.lng, props.coordinates.lat]);
+    }
+  }, [props.coordinates]);
+
+  useEffect(() => {
+    if (!mapRef.current) return;
+
+    const map = mapRef.current;
+
+    map.setZoom(props.zoom || defaultZoom);
+  }, [props.zoom]);
+
+  return (
+    <div
+      className={props.className}
+      ref={mapContainerRef}
+      id="map"
+      style={props.style}
+    ></div>
+  );
+};
 
 interface MapControlsProps {
-    // path type is temporary
-    onDone?: (path: number) => void
-    onCancel?: () => void
+  // path type is temporary
+  onDone?: (path: number) => void;
+  onCancel?: () => void;
 }
 
 const MapControls: React.FC<MapControlsProps> = (props) => {
-    return (
-        <Card
-            className="ring-transparent absolute top-0 sm:m-2 right-0 xsm:w-full sm:w-80 backdrop-blur bg-white/50"
+  return (
+    <Card className="ring-transparent absolute top-0 sm:m-2 right-0 xsm:w-full sm:w-80 backdrop-blur bg-white/50">
+      <TabGroup className="mt-1 flex justify-center">
+        <TabList variant="solid">
+          <Tab value="1">Point</Tab>
```

Comment:
maybe we should make it so that you can select a point only when the "point" tab is selected